R/repair-column-names.R

```diff
@@ -0,0 +1,19 @@
+#' Check validity/format of a data.frame or tbl.
+#'
+#' Ensure the tbl or data.frame has legitimate and optionally unique
+#' column names.
+#' @param x tbl, tbl__df, or data.frame
+#' @param unique logical (default \code{TRUE}), force column names to
+#'     be unique (\code{TRUE}) or allow duplicates (\code{FALSE});
+#'     ignored if \code{check.names} is \code{FALSE}
+#' @return data.frame, with optionally different column names
+#' @export
+repair_names <- function(x, unique=TRUE) {
+    col_names <- colnames(x)
+    if (is.null(col_names)) {
+        col_names <- rep('', ncol(x))
```

Comment:
This will give names `"X", "X.1", "X.2", "X.3", ...`.

Also, `make.names()` below will "repair" existing but non-lexical names, and also reserved words. I'm afraid this is not in the spirit of this package. I like the way matrix conversion is handled by base R:

```
> as.data.frame(diag(3))
  V1 V2 V3
1  1  0  0
2  0  1  0
3  0  0  1
```

I suggest the following:

- Empty names are set to "V1", "V2", "V3", ..., skipping already existing columns whose names match `"^V[0-9]+$"`
    - This logic might become messy and should be encapsulated in a separate function
- Duplicate names are assigned a numeric suffix (without the dot), again watch out for name clashes
    - Perhaps the logic here shares a part of the logic from the point above